src/main/java/xizi/chatbot/Parser.java

```diff
@@ -0,0 +1,96 @@
+package xizi.chatbot;
+
+import xizi.chatbot.command.ByeCommand;
+import xizi.chatbot.command.Command;
+import xizi.chatbot.command.CommandType;
+import xizi.chatbot.command.DeadlineCommand;
+import xizi.chatbot.command.DeleteCommand;
+import xizi.chatbot.command.EventCommand;
+import xizi.chatbot.command.FindCommand;
+import xizi.chatbot.command.HelpCommand;
+import xizi.chatbot.command.ListCommand;
+import xizi.chatbot.command.ListOnCommand;
+import xizi.chatbot.command.MarkCommand;
+import xizi.chatbot.command.TodoCommand;
+import xizi.chatbot.command.UnmarkCommand;
+import xizi.chatbot.task.TaskList;
+
+import java.io.IOException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * The {@code Parser} class provides utility methods for parsing user input,
+ * validating task numbers, and converting date-time strings to {@code LocalDateTime} objects.
+ * It also interprets user commands and creates the corresponding {@code Command} instances.
+ */
+public class Parser {
+    private static final DateTimeFormatter INPUT_DATE_FORMAT = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
```

Comment:
Just a lil confused, the type is of DateTimeFormatter so it sounds like its a object that has methods to help to something but the varaible is called INPUT_DATE_FORMAT which sounds like a noun and does not have any methods to do anything. Idk if you get what i mean